hawkular-alerts-api/src/main/java/org/hawkular/alerts/api/services/AlertsCriteria.java

```diff
@@ -68,6 +72,50 @@ public void setEndTime(Long endTime) {
         this.endTime = endTime;
     }
 
+    public Long getStartResolvedTime() {
+        return startResolvedTime;
+    }
+
+    /**
+     * @param startResolvedTime fetched Alerts must have resolvedTime greater than or equal to startResolvedTime
```

Comment:
I think we need more sematics here. What about alerts that have been resolved multiple times?  The same goes for the other start/end setters.